packages/react/slot/src/Slot.tsx

```diff
@@ -11,63 +11,24 @@ interface SlotProps {
 
 const Slot = React.forwardRef<HTMLElement, SlotProps>((props, forwardedRef) => {
   const { children, ...slotProps } = props;
-  const childLength = React.Children.count(children);
-
-  if (childLength === 1) {
-    return (
-      <SlotClone {...slotProps} ref={forwardedRef}>
-        {children}
-      </SlotClone>
-    );
-  }
-
-  return (
-    <>
-      {React.Children.map(children, (child) => {
-        if (React.isValidElement(child) && child.type === Slottable) {
-          return (
-            <SlotClone {...slotProps} ref={forwardedRef}>
-              {child.props.children}
-            </SlotClone>
-          );
-        }
-
-        return child;
-      })}
-    </>
-  );
-});
-
-Slot.displayName = 'Slot';
-
-/* -------------------------------------------------------------------------------------------------
- * SlotClone
- * -----------------------------------------------------------------------------------------------*/
-
-interface SlotCloneProps {
-  children: React.ReactNode;
-}
-
-const SlotClone = React.forwardRef<any, SlotCloneProps>((props, forwardedRef) => {
-  const { children, ...slotProps } = props;
-  const child = React.Children.only(children);
-
-  return React.isValidElement(child)
-    ? React.cloneElement(child, {
-        ...mergeProps(slotProps, child.props),
-        ref: composeRefs(forwardedRef, (child as any).ref),
-      })
+  return children
+    ? React.Children.only(children) && React.isValidElement(children)
```

Comment:
`React.Children.only` already does `React.isValidElement` under the hood but I have had to do it again here to hold TypeScript's hand and prevent the `children.props` code from erroring.